lang_GENERIC/analyze/Constant_propagation.ml

```diff
@@ -12,15 +12,13 @@
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
  * license.txt for more details.
 *)
-open Common
 open AST_generic
-module H = AST_generic_helpers
 module V = Visitor_AST
 
 (*****************************************************************************)
 (* Prelude *)
 (*****************************************************************************)
-(* Very basic constant propagation (no dataflow analysis involved).
+(* Flow-sensitive constant propagation.
```

Comment:
For now, I would prefer to keep the old code, so that we are backwards compatible with what we have before,
we handle things like the const keyword, the final keyword on fields in Java, etc.
Just create a Constant_propagation2.ml where you put the dataflow-based analysis and
we can call both analysis. The second one should just confirm or add more id_const_literal.

Reply:
You can add a comment at the top of Constant_propagation.ml saying this whole file should disappear at some point as we should gradually switch to the dataflow-based const analysis currently in Constant_propagation2.ml (or Constant_propagation_dataflow.ml)

Reply:
My intuition was that those cases should be correctly handled by `Dataflow_constness` but you're right, we should make sure that is the case, I could easily be missing something.

Reply:
Of course it doesn't handle fields well. I've added a TODO note to fix that later!